x-pack/legacy/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -217,25 +224,53 @@ export function App({
                     toDate: dateRange.to,
                   },
                   query: query || state.query,
-                  localQueryBarState: payload,
                 });
               }}
-              onChange={localQueryBarState => {
-                setState({ ...state, localQueryBarState });
+              filters={state.filters}
+              onFiltersUpdated={filters => {
+                data.filter.filterManager.setFilters(filters);
               }}
-              isDirty={isLocalStateDirty(state.localQueryBarState, state.query, state.dateRange)}
               appName={'lens'}
-              indexPatterns={state.indexPatternTitles}
+              indexPatterns={state.indexPatterns}
               store={store}
+              showSearchBar={true}
               showDatePicker={true}
               showQueryInput={true}
-              query={state.localQueryBarState.query}
-              dateRangeFrom={
-                state.localQueryBarState.dateRange && state.localQueryBarState.dateRange.from
-              }
-              dateRangeTo={
-                state.localQueryBarState.dateRange && state.localQueryBarState.dateRange.to
-              }
+              showFilterBar={true}
+              showSaveQuery={true /* TODO: Use permissions */}
+              savedQuery={state.savedQuery}
+              onSaved={savedQuery => {
+                setState({ ...state, savedQuery });
+              }}
+              onSavedQueryUpdated={savedQuery => {
```

Comment:
We need to make a copy of the saved query here so that if any changes made to an active saved query (one that's loaded) can be cancelled by clicking on that same saved query in the saved query management popover list.

Reply:
You also need to take care of the time filter changes _if a saved query has one_.
`                setState({
                  ...state,
                  savedQuery,
                  dateRangeFrom: savedQuery.attributes.timefilter.from,
                  dateRangeTo: savedQuery.attributes.timefilter.to
                });`
I'm not sure if you're using the refresh interval too.